### ISSUE TITLE
Merlin Locate error

### ISSUE THREAD

**Issue by oleksandr-oksenenko** [ocaml/ocaml-lsp#1187]
I'm trying to learn OCaml created a simple project using `dune init proj`, added core as a dependency and trying to go to the definition of `Set` in the following code:
```ocaml
open Core

module S = Set

let () = print_endline "Hello, World!"
```
and it fails with the following error:
```
'Locate' query to merlin returned error: File_not_found: Several source files in your path have the same name, and merlin doesn't know which is the right one: /Users/oleksandr.oksenenko/.opam/5.1.0/lib/core/set.mli, /Users/oleksandr.oksenenko/.opam/5.1.0/lib/core/set.ml, /Users/oleksandr.oksenenko/.opam/5.1.0/lib/base/set.mli, /Users/oleksandr.oksenenko/.opam/5.1.0/lib/base/set.ml, /Users/oleksandr.oksenenko/.opam/5.1.0/lib/ocaml/set.mli, /Users/oleksandr.oksenenko/.opam/5.1.0/lib/ocaml/set.ml"
```

I found the [following issue](https://github.com/ocaml/merlin/issues/894) in Merlin, however, it seems to be fixed a long time ago.

I'm using latest versions of neovim and ocaml-lsp on OCaml 5.1.0, however, i encountered the same issue with 4.14.1.

Am i missing something?

**Comment by ulugbekna** [comment]
@voodoos do you happen to know what could cause this?

**Comment by oleksandr-oksenenko** [comment]
@voodoos thanks for reply! Please let me know if i can provide any additional details.
The project builds and runs just fine:
```
~/code/ocaml_playground_2 via 🐫 v5.1.0 (5.1.0) took 16s 
❯ dune build @all
~/code/ocaml_playground_2 via 🐫 v5.1.0 (5.1.0) 
❯ dune exec -- bin/main.exe                 
Hello, World!
```

**Comment by voodoos** [comment]
Could you give me the output of `ls _build/default/bin/.merlin-conf` and `ls _build/default/bin/.*.objs/byte` ?

**Comment by oleksandr-oksenenko** [comment]
The first one produces
```
❯ ls _build/default/bin/.merlin-conf 
exe-main
```
I updated the second command a bit to find what you want (i think, the dir is called `.main.eobjs/`):
```
❯ ls _build/default/bin/.*objs/byte
dune__exe__Main.cmi  dune__exe__Main.cmti
```

**Comment by oleksandr-oksenenko** [comment]
Hm, I ran `dune build @check` and it fails with the same error. The contents of the `byte` directory: 
```
❯ ls _build/default/bin/.main.eobjs/byte
dune__exe__Main.cmi  dune__exe__Main.cmo  dune__exe__Main.cmt  dune__exe__Main.cmti
```

**Comment by voodoos** [comment]
Right, I was able to reproduce, I will have a look. Thanks for all the information !

**Comment by oleksandr-oksenenko** [comment]
Awesome, thanks!
JFYI it happens not only with `core`, for instance i had the same issue with `cohttp` modules.

**Comment by rachitnigam** [comment]
@voodoos Was there a fix for this issue? I'm attempting to follow the "getting started" instructions with OCaml and Emacs and am running into the same issue. I've tried a bunch of things including using `dune pkg lock` in hopes that it would pull in the right dependencies into the local folder.
